packages/core/core/src/services/document-service/repository.ts

```diff
@@ -393,5 +393,11 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
     publish: hasDraftAndPublish ? wrapInTransaction(publish) : (undefined as any),
     unpublish: hasDraftAndPublish ? wrapInTransaction(unpublish) : (undefined as any),
     discardDraft: hasDraftAndPublish ? wrapInTransaction(discardDraft) : (undefined as any),
+    /**
+     * @internal
+     * These methods are exposed for use within document service middlewares
+     */
+    updateComponents,
+    omitComponentData,
```

Comment:
could you mark both as internal? 
```suggestion
    /**
     * @internal
     * Exposed for use within document service middlewares
     */
    updateComponents,
        /**
     * @internal
     * Exposed for use within document service middlewares
     */
    omitComponentData,
```

Reply:
Have you discussed with alex alternative apis to not rely on exposing internal methods?

Reply:
Yes and I'll handle it once that's merged :)